src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (head == null) {
+            head = new Node<>(null, value, null);
+            tail = head;
+        } else {
+            tail.next = new Node<>(tail, value, null);
+            tail = tail.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkIndex(index, index > size);
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> nodeByIndex = node(index);
+            Node<T> prevNode = nodeByIndex.previous;
+            Node<T> toAddNode = new Node<>(prevNode, value, nodeByIndex);
+            nodeByIndex.previous = toAddNode;
+            if (prevNode == null) {
+                head = toAddNode;
+            } else {
+                prevNode.next = toAddNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index, index >= size);
+        Node<T> node = head;
+        for (int i = 0; i < index; i++) {
+            node = node.next;
+        }
+        return node.value;
```

Comment:
```suggestion
        checkIndex(index, index >= size);
        Node<T> node = head;
        for (int i = 0; i < index; i++) {
            node = node.next;
        }
        return node.value;
```
maybe you can use your function "node()" here?